app/api.js

```diff
@@ -1,19 +1,21 @@
 const SITE_ENDPOINT = '/site';
 
-export function fetchSiteState() {
-  return fetch(SITE_ENDPOINT, {
-    credentials: 'same-origin',
-  })
-  .then(response => {
-    if (response.status === 200) return response.json();
-    throw new Error('Unable to load site content from server');
-  })
-  .catch(error => { throw error; });
+function assertStatusOK(response) {
+  if (response.status >= 200 && response.status < 300) {
```

Comment:
If the [`fetch` global`](https://github.com/redbadger/react.london/pull/90/files#r69397273) is based on the [fetch API](https://developer.mozilla.org/en/docs/Web/API/Fetch_API), you can use [`response.ok`](https://developer.mozilla.org/en-US/docs/Web/API/Response/ok) as a shorthand instead of checking the range more imperatively, like you're doing here.